packages/react-utils/src/components/TableLayout/index.tsx

```diff
@@ -86,6 +88,13 @@ export function TableLayout<T extends object = Dictionary>(props: TableProps<T>)
       }),
   };
 
+  // auto append key into data if not provided
+  const data: T[] = useMemo(() => {
+    return datasource.map((e, index) => {
+      return { ...e, key: e.key ?? index };
+    });
+  }, [datasource]);
+
```

Comment:
@ashfahan do we have to remove the keys from existing datasources, considering this adds them optionally when one is not provided? 
- what's the chance that we will have conflicting keys in a page when we add more than one table?
- Also am generally iffy about moving alot of the data logic to this component, it will only increase coupling btwn the data and the component, don't you think?

Reply:
we don't really have to remove keys from existing data base but i did it to improve our code and remove the need of it. 

the tablelayout is already configured to have this as optional feature. like if you add the keys it will use them and if you dont it will append it own keys.

regarding the coupling of data and component the only features we are adding to this single component are all optional (you can override them as you want them) and is to automate alot of stuff we do on daily basis into the component they are related to. i could have created separate component or have made it more modular but the data we are dealing with is directly related to this component (Table layout)

Reply:
> we don't really have to remove keys from existing data base but i did it to improve our code and remove the need of it.

ok, kind of adds surplus changes to the pr, I will let it slide as long as the issue below is addressed

About this: `what's the chance that we will have conflicting keys in a page when we add more than one table?` I think we need to namespace the kyes in someway to make sure they are unique per table, otherwise I do not think this improves that part of the codebase as you intended.

Reply:
new realization is that keys are to do with elements identity, and the uniqueness of the identity does not need apply across different arrays. 

Also we should try avoid using indexes as keys, however, namespacing the keys like `${someRandom} ${index}` is really not different to `${index}`. 

ideally the keys should be derived from a unique aspect of the data like the objects id. while I am not against this change I think the existing keys in some places like [this](https://github.com/opensrp/web/pull/742/files#r667789765) should remain as they are.